mdn/src/test/scala/org/apache/james/mdn/MDNReportParserTest.scala

```diff
@@ -0,0 +1,202 @@
+    val parser = new MDNReportParser(disposition)
+    val result = parser.dispositionField.run()
+    assertThat(result.isSuccess).isTrue
+    assertThat(result.get).isEqualTo(expected)
+  }
+
+  @Test
+  def dispositionFieldShouldParseWhenAutomaticManualWithDisplayedType(): Unit = {
+    val disposition = "Disposition: automatic-action/MDN-sent-manually;processed"
+    val expected = Disposition.builder.actionMode(DispositionActionMode.Automatic).sendingMode(DispositionSendingMode.Manual).`type`(DispositionType.Processed).build
+    val parser = new MDNReportParser(disposition)
+    val result = parser.dispositionField.run()
+    assertThat(result.isSuccess).isTrue
+    assertThat(result.get).isEqualTo(expected)
+  }
+
+  @Test
+  def dispositionFieldShouldParseWhenDeletedType(): Unit = {
+    val disposition = "Disposition: automatic-action/MDN-sent-manually;deleted"
+    val expected = Disposition.builder.actionMode(DispositionActionMode.Automatic).sendingMode(DispositionSendingMode.Manual).`type`(DispositionType.Deleted).build
+    val parser = new MDNReportParser(disposition)
+    val result = parser.dispositionField.run()
+    assertThat(result.isSuccess).isTrue
+    assertThat(result.get).isEqualTo(expected)
+  }
+
+  @Test
+  def dispositionFieldShouldParseWhenDispatchedType(): Unit = {
+    val disposition = "Disposition: automatic-action/MDN-sent-manually;dispatched"
+    val expected = Disposition.builder.actionMode(DispositionActionMode.Automatic).sendingMode(DispositionSendingMode.Manual).`type`(DispositionType.Dispatched).build
+    val parser = new MDNReportParser(disposition)
+    val result = parser.dispositionField.run()
+    assertThat(result.isSuccess).isTrue
+    assertThat(result.get).isEqualTo(expected)
+  }
+
+  @Test
+  def dispositionFieldShouldParseWhenDisplayedType(): Unit = {
+    val disposition = "Disposition: automatic-action/MDN-sent-manually;displayed"
+    val expected = Disposition.builder.actionMode(DispositionActionMode.Automatic).sendingMode(DispositionSendingMode.Manual).`type`(DispositionType.Displayed).build
+    val parser = new MDNReportParser(disposition)
+    val result = parser.dispositionField.run()
+    assertThat(result.isSuccess).isTrue
+    assertThat(result.get).isEqualTo(expected)
+  }
+
+  @Test
+  def errorFieldShouldParse(): Unit = {
+    val error = "Error: Message1"
+    val parser = new MDNReportParser(error)
+    val result = parser.errorField.run()
+    assertThat(result.isSuccess).isTrue
+    assertThat(result.get).isEqualTo(new Error(Text.fromRawText("Message1")))
+  }
+
+  @Test
+  def extensionFieldShouldParse(): Unit = {
+    val extension = "X-OPENPAAS-IP: 177.177.177.77"
+    val parser = new MDNReportParser(extension)
+    val result = parser.extentionField.run()
+    assertThat(result.isSuccess).isTrue
+    assertThat(result.get).isEqualTo(ExtensionField.builder.fieldName("X-OPENPAAS-IP").rawValue(" 177.177.177.77").build)
+  }
+}
```

Comment:
We have tests for any valid input however I did not spotted anything for parsing errors.

Would it make sense to invest a few hours writing them? Would it have helped catching some of Raphael comments?

Reply:
I don't know i think Raphael. I was mainly backport of JAMES-3200 and i think Raphael judged it was not worth writing them.

Reply:
Ok with me then ;-)